examples/plot_score_alg.py

```diff
@@ -0,0 +1,210 @@
+        prediction_model=MyCustomModel(1),
+        noise_model=gcm.ScipyDistribution(stats.binom, p=0.8, n=1),
+    ),
+)
+causal_model.set_causal_mechanism(
+    "w",
+    gcm.AdditiveNoiseModel(
+        prediction_model=MyCustomModel(1),
+        noise_model=gcm.ScipyDistribution(stats.binom, p=0.5, n=1),
+    ),
+)
+
+# Fit here would not really fit parameters, since we don't do anything in the fit method.
+# Here, we only need this to ensure that each FCM has the correct local hash (i.e., we
+# get an inconsistency error if we would modify the graph afterwards without updating
+# the FCMs). Having an empty data set is a small workaround, since all models are
+# pre-defined.
+gcm.fit(causal_model, pd.DataFrame(columns=["x", "y", "z", "w"]))
+
+# sample the observational data
+data = gcm.draw_samples(causal_model, num_samples=500)
+
+print(data.head())
+print(pd.Series({col: data[col].unique() for col in data}))
+draw(G)
+
+# %%
+# Define the context
+# ------------------
+# In PyWhy, we introduce the :class:`context.Context` class, which should be a departure from
+# "data-first causal discovery," where users provide data as the primary input to a
+# discovery algorithm. This problem with this approach is that it encourages novice
+# users to see the algorithm as a philosopher's stone that converts data to causal
+# relationships. With this mindset, users tend to surrender the task of providing
+# domain-specific assumptions that enable identifiability to the algorithm. In
+# contrast, PyWhy's key strength is how it guides users to specifying domain
+# assumptions up front (in the form of a DAG) before the data is added, and
+# then addresses identifiability given those assumptions and data. In this sense,
+# the Context class houses both data, apriori assumptions and other relevant data
+# that may be used in downstream structure learning algorithms.
+context = make_context().variables(data=data).build()
+
+# Alternatively, one could say specify some fixed edges.
+# Note that when specifying fixed edges, the resulting graph that is
+# learned is not necessarily a "pure CPDAG". In that, there are more
+# constraints than just the conditional independences. Therefore, one
+# should use caution when specifying fixed edges if they are interested
+# in leveraging ID or estimation algorithms that assume the learned
+# structure is a "pure CPDAG".
+
+# .. code-block::Python
+#   included_edges = nx.Graph([('x', 'y')])
+#   context = make_context().edges(include=included_edges).build()
+
+# %%
+# Run structure learning algorithm
+# --------------------------------
+# Now we are ready to run the SCORE algorithm. The method performs inference
+# in two phases. First it estimates the topological order of the nodes in the
+# graphs. This is done iteratively according to the following procedure:
+#
+# 1. SCORE estimates the Hessian of the logarithm of :math:`p(x)`,
+# with :math:`p(x)` the joint distribution of the nodes in the graph.
```

Comment:
```suggestion
# 1. SCORE estimates the Hessian of the logarithm of :math:`p(V)`,
# where :math:`p(V)` is the joint distribution of the nodes in the graph (``V`` are the nodes in the graph).
```

Is this right? the Hessian is estimated over the joint distribution of all nodes in the graph? So it is a $d \times d$ matrix where d is the number of nodes?

Reply:
I use `V` just to stick with convention, where `x` would be a singleton

Reply:
> Is this right? the Hessian is estimated over the joint distribution of all nodes in the graph? So it is a 
 matrix where d is the number of nodes?

Exactly